apps/docs-snippets/src/guide/provider/querying-the-chain.test.ts

```diff
@@ -125,4 +126,108 @@ describe('querying the chain', () => {
     expect(message?.nonce).toEqual(nonce);
     // #endregion getMessageByNonce
   });
+
+  it('can getMessage', async () => {
+    // #region Message-getMessages
+    // #import { TestMessage, launchTestNode };
+
+    // Creates a test message with an amount of 100
+    const testMessage = new TestMessage({ amount: 100 });
+
+    // Launches a test node with the test message configured
+    using launched = await launchTestNode({ walletsConfig: { messages: [testMessage] } });
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    // Retrieves messages from the wallet
+    const [message] = await wallet.getMessages();
+    // #endregion Message-getMessages
+
+    expect(message.nonce).toEqual(testMessage.nonce);
+  });
+
+  it('can getMessageProof with blockId', async () => {
+    // #region Message-getMessageProof-blockId
+    // #import { launchTestNode, TransactionResultMessageOutReceipt };
+
+    // Launches a test node with two wallets established
+    using launched = await launchTestNode({
+      walletsConfig: {
+        count: 2,
+      },
+    });
+
+    const {
+      wallets: [sender, recipient],
+      provider,
+    } = launched;
+
+    // Defines the recipient address
+    const recipientAddress = recipient.address.toB256();
+
+    // Performs a withdrawal transaction from sender to recipient
+    const tx = await sender.withdrawToBaseLayer(recipientAddress, 100);
```

Comment:
Why are we performing this tx twice, like again on line 175?

Reply:
To get the nonce, I need to make another transaction to create a block. What I can do is set the `recipient` to be the `sender` in this second transaction.